pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -370,3 +371,116 @@ func (h *appHandler) applyHelmModuleResources(ctx context.Context, comp *v1alpha
 	klog.InfoS("Apply a HelmRelease", "namespace", release.GetNamespace(), "name", release.GetName())
 	return nil
 }
+
+// handleResourceTracker check the namesapce of  all components and traits, if the namespace is different with application, the tracker will own them
+func (h *appHandler) handleResourceTracker(components []*v1alpha2.Component, ac *v1alpha2.ApplicationConfiguration, appNs string, tracker *v1alpha2.ResourceTracker) error {
+	ref := metav1.NewControllerRef(tracker, v1alpha2.ResourceTrackerKindVersionKind)
+	for i := 0; i < len(components); i++ {
+		c := components[i]
+		u, err := oamutil.RawExtension2Unstructured(&c.Spec.Workload)
+		if err != nil {
+			return err
+		}
+		if len(u.GetNamespace()) != 0 && u.GetNamespace() != appNs {
```

Comment:
please write a function for this check, so we can align the checking logic everywhere.